feat: allow math expressions for time and duration (#24453)
open-telemetry/opentelemetry-collector-contrib: pkg/ottl/README.md

Math Expressions represent arithmetic calculations.  They support `+`, `-`, `*`, and `/`, along with `()` for grouping.

> Math Expressions currently support `int64` and`float64`. 
> Math Expressions also support `time.Time` and `time.Duration`, but only with `+` and `-` and only using the following rules: 

Reviewer: ```suggestion
Math Expressions currently support `int64`, `float64`, `time.Time` and `time.Duration`. 
For `time.Time` and `time.Duration`, only `+` and `-` are supported with the following rules: 
```

  - A `time.Time` `-` a `time.Time` yields a `time.Duration`.
  - A `time.Duration` `+` a `time.Time` yields a `time.Time`. 

>   - A `time.Time` `-`  a `time.Duration` yields a `time.Time`.

Reviewer 2: ```suggestion
  - A `time.Time` `+`  a `time.Duration` yields a `time.Time`.
  - A `time.Time` `-`  a `time.Duration` yields a `time.Time`.
```

Could we add `time + duration`? I get that the commutative case is already covered here, but ideally users don't have to think about the order.

Author: Sounds great!
